docs/guides/admin/docs/configuration/lifecycle-management.md

```diff
@@ -0,0 +1,54 @@
+LifeCycle Management
+====================
+
+The goal of LifeCycle Management is to allows users to automate administrative tasks that are specific to their use
+case. For example:
+
+" We want to automatically retract videos of lecture recordings that have been uploaded during the winter semester 2024
+at the end of the semester. Lecture recordings are always organized in a series. Conferences should be excluded from
+this, as well as all lectures by the VIP Albert Einstein."
+
+To that end, user can create LifeCycle Policies. They allow users to define
+
+- what should happen (action),
+- to what it should happen to (target/object) and
+- when it should happen (time).
+
+The different components of a policy are described below in greater detail.
+
+## Policy Action
+
+For now there is only one action that can be specified, called "START_WORKFLOW". “START_WORKFLOW” takes a workflow
+definition ID (and workflow configuration) to define which workflow should be run. For example, in order to retract
```

Comment:
For each event, a so called "policy task" is created. Should the task run into an exception for any reason, like no workflow definition, it will log the exception and set itself to "FAILED" (which effectively means it won't be retried).

The code does not currently check if the user who created the lifecycle policy has access to the filtered events. Maybe it should?

Reply:
On one hand it might make sense to ensure that a user can't create policies for things they don't have access too, but it's also reasonable to suspect that the time between policy creation and policy execution might be long enough that their roles might change.  I'd be tempted to check at both?  Definitely need to check at execution time, but it'll probably make more sense to check at schedule time for UI uses...

Regardless, docs for the workflow definition bits, and whatever happens to address the above should be added.

Reply:
Feels like this requires a concept of "policy ownership" to decide which entities a policy can affect, but that seems not necessarily trivial?
- Who should own the policy? The creator?
- Even then, if someone else than the creator with write access to the policy makes changes to it, they could potentially still affect events that they would normally not be allowed to edit. Is that a problem?

Reply:
I would avoid creating policy ownership IMO.  It raises, as you point out, a whole bunch of questions with no obvious answers.  If that means there is more in terms of undefined behaviour (like being able to create policies against events you don't have access to at schedule time) then so be it.  Adding ownership here doesn't get us much, especially since we more or less have to check ACLs at runtime regardless.  A simple 'does the ACL listed in the policy match the event' is probably enough almost all usecases.